llvm/test/tools/sycl-post-link/sycl-esimd/basic-sycl-esimd-split.ll

```diff
@@ -0,0 +1,44 @@
+; RUN: sycl-post-link -split=auto -S %s -o %t.table
+; RUN: FileCheck %s -input-file=%t.table
+; RUN: FileCheck %s -input-file=%t_0.ll --check-prefixes CHECK-SYCL-IR
+; RUN: FileCheck %s -input-file=%t_esimd_0.ll --check-prefixes CHECK-ESIMD-IR
+
+; This is basic test of splitting SYCL and ESIMD kernels into separate
+; modules.
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-linux-sycldevice"
+
+declare dso_local spir_func i64 @_Z28__spirv_GlobalInvocationId_xv()
+
+define dso_local spir_kernel void @ESIMD_kernel() #0 !sycl_explicit_simd !3{
+entry:
+  %call = tail call spir_func i64 @_Z28__spirv_GlobalInvocationId_xv()
+  ret void
+}
+
+define dso_local spir_kernel void @SYCL_kernel() #1 {
+entry:
+  %call = tail call spir_func i64 @_Z28__spirv_GlobalInvocationId_xv()
+  ret void
+}
+
+attributes #0 = { "sycl-module-id"="a.cpp" }
+attributes #1 = { "sycl-module-id"="a.cpp" }
+
+!llvm.module.flags = !{!0}
+!opencl.spir.version = !{!1}
+!spirv.Source = !{!2}
+
+!0 = !{i32 1, !"wchar_size", i32 4}
+!1 = !{i32 1, i32 2}
+!2 = !{i32 0, i32 100000}
+!3 = !{}
+
+; CHECK: [Code|Properties]
+; CHECK: {{.*}}_0.ll|{{.*}}_0.prop
+; CHECK: {{.*}}_esimd_0.ll|{{.*}}_esimd_0.prop
+
+; CHECK-SYCL-IR: define dso_local spir_kernel void @SYCL_kernel()
```

Comment:
I expect `_Z28__spirv_GlobalInvocationId_xv` to be part of both modules (cloned) - can you please add a check for that?

Reply:
Done